### PR TITLE
fix: set app icon for all targets

### PR DESCRIPTION
**Pull request by aesteves900**
GitHub Issue (If applicable): #1038 

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by nickrandolph — COMMENTED** [review]
The logo.ico file needs to be regenerated as it looks like it's got a watermark applied to top right corner

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Skia.Gtk/MyExtensionsApp.Skia.Gtk.csproj [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 		<TargetFramework>net6.0</TargetFramework>
+		<ApplicationIcon>Images\logo.ico</ApplicationIcon>
 	</PropertyGroup>
 	<!--#if winAppSdk-->
 	<ItemGroup Condition="exists('Platforms\Windows')">
 		<EmbeddedResource Include="Platforms\Windows\Package.appxmanifest" LogicalName="Package.appxmanifest" />
 		<Content Include="Platforms\Windows\Images\StoreLogo.png" />
 	</ItemGroup>
+	<ItemGroup>
+	  <None Remove="Images\logo.ico" />
```
Comment (nickrandolph) [inline src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Skia.Gtk/MyExtensionsApp.Skia.Gtk.csproj:14]: This line isn't required

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Skia.Linux.FrameBuffer/MyExtensionsApp.Skia.Linux.FrameBuffer.csproj [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 		<TargetFramework>net6.0</TargetFramework>
+		<ApplicationIcon>Images\logo.ico</ApplicationIcon>
 	</PropertyGroup>
 	<!--#if winAppSdk-->
 	<ItemGroup Condition="exists('Platforms\Windows')">
 		<EmbeddedResource Include="Platforms\Windows\Package.appxmanifest" LogicalName="Package.appxmanifest" />
 		<Content Include="Platforms\Windows\Images\StoreLogo.png" />
 	</ItemGroup>
+	<ItemGroup>
+	  <None Remove="Images\logo.ico" />
```
Comment (nickrandolph) [inline src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Skia.Linux.FrameBuffer/MyExtensionsApp.Skia.Linux.FrameBuffer.csproj:14]: Not required

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (nickrandolph) [inline src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json:1]: @dansiegel do we want to include launchSettings.json for the WASM target?

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Windows/MyExtensionsApp.Windows.csproj [hunk 4]
```diff
@@ -16,10 +16,12 @@
 		<!-- This bundles the .NET Core libraries (Uncomment for packaged builds)  -->
 		<SelfContained>true</SelfContained>
 		<DefaultLanguage>en</DefaultLanguage>
+		<ApplicationIcon>Images\logo.ico</ApplicationIcon>
```
Comment (nickrandolph) [inline src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Windows/MyExtensionsApp.Windows.csproj:19]: ApplicationIcon isn't required or useful for Windows target as icon etc is defined by package.appxmanifest file

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json (hunk 3 above)
Comment (dansiegel) [reply src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json:1]: afaik there shouldn't be a single change for the launchSettings.json in this PR

**Comment by aesteves900** [comment]
> The logo.ico file needs to be regenerated as it looks like it's got a watermark applied to top right corner

Done!

File: src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json (hunk 3 above)
Comment (dansiegel) [reply src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.Wasm/Properties/launchSettings.json:1]: be sure to remove this. It is not related to the issue and is fixed in another PR

**Pull request closed without merging**